lib/servers/evPump/evPumpServer.py

```diff
@@ -0,0 +1,227 @@
+instancename = eVPump
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+[shutdown]
+message = 987654321
+timeout = 20
+### END NODE INFO
+"""
+
+'''
+Created on Mar 25, 2016
+
+@author: Anthony Ransford
+'''
+
+UPDATECURR = 150327
+UPDATEPOW = 114327
+UPDATETMP = 153422
+UPDATESTAT = 356575
+
+from common.lib.servers.serialdeviceserver import SerialDeviceServer, setting, inlineCallbacks, SerialDeviceError, SerialConnectionError
+from twisted.internet import reactor
+from labrad.server import Signal
+from labrad import types as T
+from twisted.internet.defer import returnValue
+from labrad.support import getNodeName
+from labrad.units import WithUnit as U
+
+SERVERNAME = 'eVPump'
+TIMEOUT = 1.0
+BAUDRATE = 115200
+
+class eVPump(SerialDeviceServer):
+    name = SERVERNAME
+    regKey = 'eVPump'
+    port = None
+    serNode = getNodeName()
+    timeout = T.Value(TIMEOUT,'s')
+    temperature = None
+    power = None
+    current = None
+    status = None
+
+    currentchanged = Signal(UPDATECURR, 'signal: current changed', 'v')
+    powerchanged = Signal(UPDATEPOW, 'signal: power changed', 'v')
+    temperaturechanged = Signal(UPDATETMP, 'signal: temp changed', 'v')
+    statuschanged = Signal(UPDATESTAT, 'signal: stat changed', 's')
+    
+    @inlineCallbacks
+    def initServer( self ):
+        if not self.regKey or not self.serNode: 
+            raise SerialDeviceError( 'Must define regKey and serNode attributes' )
+        port = yield self.getPortFromReg( self.regKey )
+        self.port = port
+        print port
+        try:
+            serStr = yield self.findSerial( self.serNode )
+            self.initSerial( serStr, port, baudrate = BAUDRATE )
+        except SerialConnectionError, e:
+            self.ser = None
+            if e.code == 0:
+                print 'Could not find serial server for node: %s' % self.serNode
+                print 'Please start correct serial server'
+            elif e.code == 1:
+                print 'Error opening serial connection'
+                print 'Check set up and restart serial server'
+            else: raise
+        self.measure_pump()
+
+    @setting(1, 'toggle laser', value = 'b')
+    def toggle_laser(self, c, value):
+        if value:
+            yield self.ser.write_line('ON')
+        else:
+            yield self.ser.write_line('OFF')
+            
+    @setting(2, 'toggle shutter', value = 'b')
+    def toggle_shutter(self, c, value):
+        if value:
+            yield self.ser.write_line('SHT:1')
+        else:
+            yield self.ser.write_line('SHT:0')
+            
+    @setting(3, 'set power', value = 'v[W]')
+    def set_power(self, c, value):
+        value = str(value['W'])
+        yield self.ser.write_line('P:' + value)
+        
+    @setting(4, 'read power', returns = 'v[W]')
+    def read_power(self,c):
+        yield None
+        returnValue(self.power)
+        
+    @setting(5, 'set current', value = 'v[A]')
+    def set_current(self, c, value):
```

Comment:
If possible I would bundle set_current and read_current into one setting current, where if a value is supplied it sets the value, otherwise it gets the value.